src/toucan2/jdbc/read.clj

```diff
@@ -138,10 +138,12 @@
 (defn ^:no-doc make-i->thunk
   "Given a connection `conn`, a `model` and a result set `rset`, return a function which given a column number `i` returns
   a thunk that retrieves the column value of the current row from the result set."
-  [conn model rset]
-  (comp (memoize (fn [i]
-                   (make-column-thunk conn model rset i)))
-        int))
+  [conn model ^ResultSet rset]
+  (let [n (.getColumnCount (.getMetaData rset))
+        thunks (vec (for [i (range n)]
+                      (delay (make-column-thunk conn model rset (inc i)))))]
```

Comment:
Probably a slight performance improvement to use `mapv` here to avoid creating the lazy seq, right?